src/fights.ts

```diff
@@ -260,10 +262,52 @@ function embezzlerSetup() {
   if (have($item`ice sculpture`) && !get("iceSculptureMonster")) {
     visitUrl(`desc_item.php?whichitem=${$item`ice sculpture`.descid}`, false, false);
   }
+
+  if (doingExtrovermectin()) {
+    if (
+      have($skill`Transcendent Olfaction`) &&
+      (!have($effect`On the Trail`) || get("olfactedMonster") !== $monster`crate`)
+    ) {
+      if (have($effect`On the Trail`)) uneffect($effect`On the Trail`);
+      const run = findRun() ?? ltbRun;
+      const macro = Macro.trySkill($skill`Transcendent Olfaction`)
+        .trySkill($skill`Offer Latte to Opponent`)
+        .externalIf(
+          get("_gallapagosMonster") !== $monster`crate` && have($skill`Gallapagosian Mating Call`),
+          Macro.trySkill($skill`Gallapagosian Mating Call`)
+        )
+        .step(run.macro);
+
+      new Requirement(["100 Monster Level"], {
+        forceEquip: $items`latte lovers member's mug`.filter((item) => have(item)),
+      })
+        .merge(run.requirement ? run.requirement : new Requirement([], {}))
+        .maximize();
+      useFamiliar(freeFightFamiliar());
+      if (run.prepare) run.prepare();
+      adventureMacro(
+        $location`Noob Cave`,
+        Macro.if_($monster`crate`, macro)
+          .if_($monster`time-spinner prank`, Macro.kill())
```

Comment:
Saw this time-spinner prank check few places, it would probably be better suited to be handed in the main() function of combat.ts similar to tentacles.

Reply:
Declaring that out of scope of this PR, but very good idea. That won't work for our autoattack macros--a silly fix for this would be to add it to the constructor function for our specialized Macro class, but also I'm _pretty_ sure we only use this sort of thing in adventureMacro rather than adventureMacroAuto anyways